starknet_devnet/starknet_wrapper.py

```diff
@@ -53,13 +31,13 @@ def __init__(self, origin):
         self.__address2contract_wrapper: Dict[int, ContractWrapper] = {}
         """Maps contract address to contract wrapper."""
 
-        self.__transaction_wrappers: List[TransactionWrapper] = []
-        """A chronological list of transaction_wrappers."""
+        self.__transaction_wrappers: Dict[int, TransactionWrapper] = {}
+        """Maps transaction hash to transaction wrapper."""
 
         self.__hash2block = {}
         """Maps block hash to block."""
 
-        self.__own_blocks = {}
+        self.__num2block = {}
```

Comment:
You can put Dict[int, Dict] here (I know Dict is very wild card, but that's better than nothing)